clippy_lints/src/attrs/useless_attribute.rs

```diff
@@ -25,6 +25,10 @@ pub(super) fn check(cx: &LateContext<'_>, item: &Item<'_>, attrs: &[Attribute])
                                 || is_word(lint, sym!(unreachable_pub))
                                 || is_word(lint, sym!(unused))
                                 || is_word(lint, sym!(unused_import_braces))
+                                || is_word(lint, sym!(unused_braces))
```

Comment:
At this point, shouldn't we do something like:

```rust
if lint.si_word() && let Some(ident) = lint.ident() && matches!(ident.name, sym::deprecated | sym!(unreachable_pub) | ...)
```

?

Reply:
RIP:

```rust
error[E0164]: expected tuple struct or tuple variant, found associated function `rustc_span::symbol::Symbol::intern`
  --> clippy_lints/src/attrs/useless_attribute.rs:29:37
   |
29 | ...                   sym!(ambiguous_glob_reexports)
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `fn` calls are not allowed in patterns
   |
   = help: for more information, visit https://doc.rust-lang.org/book/ch18-00-patterns.html
   = note: this error originates in the macro `sym` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Then we can use `ident.name.as_str()` and match on strings. :)